File: typescript/src/booking/domain/primitive/giftCardReference.ts
// Riferimento opaco a una gift card gestita dal BC `giftcard`.
// Il BC `booking` non conosce la struttura interna di `giftcard.GiftCardId`:
// utilizza solo questo value object per indicare a quale gift card si riferisce una prenotazione.

import { requireArgument } from '@/common/utils/requireArgument.js';
import { Uuid } from '@/common/domain/primitive/uuid.js';

export class GiftCardReference {
    readonly value: Uuid;

    constructor(value: Uuid) {
        requireArgument(value, 'value');
        this.value = value;
    }
}

File: java/src/main/java/it/giannibombelli/wsc2026/booking/domain/primitive/GiftCardReference.java
package it.giannibombelli.wsc2026.booking.domain.primitive;

import it.giannibombelli.wsc2026.common.utils.Require;

import java.util.UUID;

/**
 * Riferimento opaco a una gift card gestita dal BC {@code giftcard}.
 * Il BC {@code booking} non conosce la struttura interna di {@code giftcard.GiftCardId}:
 * utilizza solo questo value object per indicare a quale gift card si riferisce una prenotazione.
 */
public record GiftCardReference(UUID value) {
    public GiftCardReference {
        Require.requireArgument(value, "value");
    }
}
